### PR TITLE
WFLY-12759 Extract microprofile-config galleon layer from observability

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.jboss.org/browse/WFLY-12759

Microprofile specs like OpenAPI don't require metrics nor health, suggesting the microprofile layers should be more granular.

### CODE REVIEW

File: testsuite/integration/microprofile-tck/config/pom.xml [hunk 1]
```diff
@@ -173,7 +173,7 @@
                                             <name>standalone.xml</name>
                                             <layers>
                                                 <layer>datasources-web-server</layer>
-                                                <layer>observability</layer>
+                                                <layer>microprofile-config</layer>
```
Comment (bstansberry) [inline testsuite/integration/microprofile-tck/config/pom.xml]: I'm not sure I want to do this as running against this combo is part of the test plan for our recommended combos in our images.

Comment (pferraro) [reply testsuite/integration/microprofile-tck/config/pom.xml]: The intention was to use a minimal profile sufficient to pass the MicroProfile Config TCK.  It also provides an opportunity to validate the microprofile-config layer.
https://github.com/wildfly/wildfly/pull/12774 will use a profile more in line with the test plan you described.
If you still disagree, I will revert this change.

Comment (bstansberry) [reply testsuite/integration/microprofile-tck/config/pom.xml]: I'd prefer not to change this right now.  Right now #12774 is an empty module.

I don't particularly disagree with the change; I just don't feel like reopening a long discussion right now when I know there will be lots of other opportunities to do it. :)

**Review by bstansberry — CHANGES_REQUESTED** [review]
This will need an entry in the testsuite/layers/pom.xml. All layers should have an entry there as that is the basic boot validation for the layer.

It also needs an update to https://github.com/wildfly/wildfly/blob/master/docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc

File: testsuite/integration/microprofile-tck/config/pom.xml (hunk 1 above)
Comment (pferraro) [reply testsuite/integration/microprofile-tck/config/pom.xml]: No problem.

**Comment by pferraro** [comment]
@bstansberry Fixed.

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
@jmesnil OK with you?

**Review by rhusar — APPROVED** [review]
Thanks for this. Looks good.

**Comment by rhusar** [comment]
Actually, would it make any sense to document this in {{docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc}}? Seems unnecessary as its basically not useful by itself.

**Comment by bstansberry** [comment]
@rhusar MicroProfile Config is usable directly by applications, so any layer at all could possibly be decorated with this one.  

It seems simpler to document them all too.

**Pull request merged**